### PR TITLE
Adding PKGBUILD for agat

### LINKED ISSUE #102: [REQUEST] add AGAT to repo
**Information**
 - Package Name: AGAT
 - Can it directly run on Linux: [yes]
 - Can it run on the wine: [NA]
 - Website: https://github.com/NBISweden/AGAT
 - Download URL: https://github.com/NBISweden/AGAT
 - Package description: Another Gtf/Gff Analysis Toolkit is suite of tools to handle gene annotations in any GTF/GFF format.
 - DOI: https://www.doi.org/10.5281/zenodo.3552717
 

**Popularity**
 - Number of citations from WOS per year: [N/A]
 - Research interests in ResearchGate: [N/A]
 - Any other reason: the package is a pretty useful in fixing augustus hint gtf file formatting generated in BRAKER2 pipeline. furthermore the package seems to be in debian-med collection which, as per limited understanding, Bioarchlinux project is trying to add to repo.

**Additional information**
i was able to compile the package using [PKGBUILD here](https://gist.github.com/kbipinkumar/779ec184aeb3c6fa2b09e592f00c8d59). the upstream recommends following dependencies (i have converted ubuntu packages recommended in upstream repo to their equivalent in archlinux ecosystem)
1. perl-bioperl
2. perl-clone
3. perl-carp-clan
4. perl-sort-naturally
5. perl-yaml-libyaml
6. perl-lwp-protocol-https
7. perl-file-sharedir-install
8. perl-file-sharedir-install
9. perl-graph
10. perl-lwp-useragent-determined
11. perl-moose
12. perl-file-share
packages 9 to 12 are present only in AUR. therefore, those also will need to be added to repo along with their dependency, if any. (Any help in this regard would be valuable)

### PR DESCRIPTION
**Pull request by kbipinkumar**
## Involved packages

 - AGAT

## Involved issue

Close #102 

## Details

- [X] Tested in the local machine (largest 16G RAM) and it is passed without any issue
- [X] Provide New Package
- [X] Fix the Packages
  - [X] PKGBUILD
  - [X] lilac.yaml
  - [ ] lilac.py
- [ ] Would like to continue to work with us

### CODE REVIEW

File: BioArchLinux/agat/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Maintainer: Bipin Kumar <bipin@ccmb.res.in>
+
+pkgname=agat
+_pkgname=AGAT
+pkgver=1.0.0
+pkgrel=1
+pkgdesc="Another Gtf/Gff Analysis Toolkit: Suite of tools to handle gene annotations in any GTF/GFF format. https://www.doi.org/10.5281/zenodo.3552717"
```
Comment (hubutui) [inline BioArchLinux/agat/PKGBUILD:7]: It would be better to make the `pkgdesc` shorter, maybe remove the doi URL?

**Comment by hubutui** [comment]
I'm not familiar with Perl packaging. Could you make sure you follow the [Perl package guidelines](https://wiki.archlinux.org/title/Perl_package_guidelines)?

**Comment by kbipinkumar** [comment]
> I'm not familiar with Perl packaging. Could you make sure you follow the [Perl package guidelines](https://wiki.archlinux.org/title/Perl_package_guidelines)?

To my best, I was following guidelines from [Perl Policy](https://wiki.archlinux.org/title/Perl_Policy#Binaries_and_scripts) from ArchWiki which refers to user installed binaries and scripts outside of CPAN

>Binaries and scripts
In order to prevent file collisions, it is important to keep binaries generated by core, vendor, and site installs separate. It is also important that the default PATH environment variable set in each users profile to search for binaries in the same order as perl's @INC path. In order to accomplish this, binaries should be installed into the following directories:
>>Core
Binaries and scripts for all core packages should be installed into /usr/bin/core_perl.
>>Vendor
Binaries and scripts for all vendor packages should be installed into /usr/bin/vendor_perl.
>>Site
Binaries and scripts for all site should default to be installed into /usr/bin/site_perl.

>The perl package should include a mechanism to adjust end-users PATH entries accordingly so that perl binaries are searched for in the following order: site, vendor, core.

as the package `agat` is a collection of perl scripts that are being installed by user outside of CPAN, it's safe to say that those can go into site path which is `/usr/bin/site_perl`

here is the output of `pacman -pQl agat-1.0.0-1-any.pkg.tar.zst`

```
agat /usr/
agat /usr/bin/
agat /usr/bin/site_perl/
agat /usr/bin/site_perl/agat
agat /usr/bin/site_perl/agat_convert_bed2gff.pl
agat /usr/bin/site_perl/agat_convert_embl2gff.pl
agat /usr/bin/site_perl/agat_sp_clipN_seqExtremities_and_fixCoordinates.pl
agat /usr/bin/site_perl/agat_sp_compare_two_BUSCOs.pl
```

you can see the `PKGBUILD` installing the script files to `/usr/bin/site_perl/` as per the perl policy.

Hope this answers your query.

File: BioArchLinux/agat/PKGBUILD (hunk 1 above)
Comment (kbipinkumar) [reply BioArchLinux/agat/PKGBUILD:7]: according to bioarchlinux  description [here](https://f1000research.com/posters/11-809) DOIs are used to query and cite packages used,

>The repository provides, where possible, a DOI in each package description. This makes it easy for users to find out more about each package’s use and methodology

i was just following that guideline. if you wish i will remove the DOI

**Comment by kbipinkumar** [comment]
> I'm not familiar with Perl packaging. Could you make sure you follow the [Perl package guidelines] (https://wiki.archlinux.org/title/Perl_package_guidelines)?

To my best, I was following guidelines from [Perl Policy](https://wiki.archlinux.org/title/Perl_Policy#Module_paths) from ArchWiki. the following part refers to perl packages/scripts via the pacman tool from repos, or from AUR

>vendor
Packaged modules, installed via the pacman tool from the core, extra, or community repositories, or built into proper Arch Linux packages from ABS/AUR PKGBUILDs.

```
/usr/lib/perl5/vendor_perl
/usr/share/perl5/vendor_perl
```

as the package `agat` is a collection of perl scripts that are being installed by user using pacman, it's safe to say that those can go into vendor path which is `/usr/bin/vendor_perl`

i have updated `PKGBUILD` to install files to `/usr/bin/vendor_perl` as in this commit 50fc2bd699cffc2fcdcdb7cee133fe02ae18ed0c

here is the output of pacman filelist query

```
pacman -pQl agat-1.0.0-1-any.pkg.tar.zst
agat /usr/
agat /usr/bin/
agat /usr/bin/vendor_perl/
agat /usr/bin/vendor_perl/agat
agat /usr/bin/vendor_perl/agat_convert_bed2gff.pl
agat /usr/bin/vendor_perl/agat_convert_embl2gff.pl
agat /usr/bin/vendor_perl/agat_convert_genscan2gff.pl
agat /usr/bin/vendor_perl/agat_convert_mfannot2gff.pl
```

you can see the `PKGBUILD` installing the script files to `/usr/bin/vendor_perl/` as per the perl policy.

Hope this answers your query.

**Comment by hubutui** [comment]
LGTM.
It would be great if you do some code format. eg, make the indent to the same spaces, make the dep one pkg per line and sorted, and so on.

**Comment by kbipinkumar** [comment]
> LGTM. It would be great if you do some code format. eg, make the indent to the same spaces, make the dep one pkg per line and sorted, and so on.

i have made some modifications per latest commit [1b6a5ba](https://github.com/BioArchLinux/Packages/pull/105/commits/1b6a5baebeb49368cc59ae5b0649f8f36cf67452). please check it out

**Comment by hubutui** [comment]
LGTM. 

1. Could you git rebase and merge all these commits into one? 
2. It would be better to change `lilac.yaml` and `PKGBUILD` to `644` permission. 
3. We could add `options=('!emptydirs' 'purge')` to `PKGBUILD` as the [guidelines](https://wiki.archlinux.org/title/Perl_package_guidelines) suggests.

**Comment by kbipinkumar** [comment]
> LGTM.
> 
> 1. Could you git rebase and merge all these commits into one?
> 2. It would be better to change `lilac.yaml` and `PKGBUILD` to `644` permission.
> 3. We could add `options=('!emptydirs' 'purge')` to `PKGBUILD` as the [guidelines](https://wiki.archlinux.org/title/Perl_package_guidelines) suggests.

Done.

**Pull request merged**